### ISSUE TITLE
Import Sterblue data into LabelFlow

### ISSUE THREAD

**Issue by nicOwlas** [labelflow/labelflow#640]
## Tasks

- [x] Create one workspace per exported organization
- [x] Create one dataset per mission
- [x] Create classes of the detection classes
- [ ] Create a membership with the role owner for every admin of the organization. nicolas.draber@sterblue.com invites the members to join the workspace.
  - [x] Create a session with `nicolas.draber@sterblue.com` to get his session token and be able to use the `inviteMember` resolver in his name
- [ ] To be done in another issue: ~~Add the organization image as an avatar (action @nicOwlas on Sterblue: adapt the logo aspect ratio to match 1x1)~~
- [ ] Create one Pro plan subscription per Workspace
  - [x] Use the LabelFlow GraphQL API (so that when you create a workspace, it creates the Stripe subscription)
- [x] Add to the `image.metadata` the image metadata in `{sterblue:{xxx}}` @action @nicOwlas 

The query info to put in every image metadata field:
```
query getImages {
  images(where: {id:"xxx"}) {
    mission {
      id
      mission {
        id
        name
      }
    }
    id
    capturedAt
    cameraMake
    cameraModel
    cameraLensMake
    cameraIso
    cameraAperture
    cameraShutterSpeed
    cameraFieldOfViewHorizontal
    cameraFieldOfViewVertical
    location {
      latitude
      longitude
      altitude
      roll
      pitch
      yaw
    }
    structures {
      id
      name
      parameters
      location {
        latitude
        longitude
        altitude
      }
      model {
        type
      }
      parents {
        id
        name
        parameters
        location {
          latitude
          longitude
          altitude
        }
        model {
          type
        }
        parents {
          id
          name
          parameters
          location {
            latitude
            longitude
            altitude
          }
          model {
            type
          }
          parents {
            id
            name
            parameters
            location {
              latitude
              longitude
              altitude
            }
            model {
              type
            }
          }
        }
      }
    }
  }
}
```
Depends on #647
Blocked by #497

**Comment by etiennedupont** [comment]
Some notes since I started investigating that:
- The naive way of doing it would be to:
   - Mirror all Sterblue users in Labelflow
   - Mirror the organizations by creating workspaces, adding all workspace memberships as well
   - Pass through the missions and create datasets.
      For each dataset / mission:
      - Query the anomaly types in the mission and create them in the dataset
      - Go through images, upload (and create) them <= this is the bottleneck
        For each image:
         - Go through detections, convert and create them

Technically, we can use the sterblue SDK for the sterblue part and try to directly use the resolvers we coded on the LabelFlow part, so we don't have to implement some heavy logic for the images with thumbnails, etc.

- Everyone linked to the workspace will access all datasets / missions, which is not the current state of permissions at Sterblue. There is a per-mission granularity at Sterblue, from what I understand. Is that an issue? If it is, there's a need to create one workspace per mission, which is a bit frustrating but I don't see many alternatives. Maybe we could make those datasets read-only, which will be a useful feature in the future anyway.
- Another problem linked to the first is that there are missions not linked to any organization, so we need a solution for that as well.
- We need to take care about duplicated organization names if there are some

**Comment by nicOwlas** [comment]
Together with @LouisAllen7 we started cleaning the Sterblue DB to facilitate the migration. The cleaning we are doing right now (should be finished tomorrow).
- we are adding the `nickname` "labelflow" to the sterblue organizations to be imported. Only those ones are to be imported. The other ones have the `nickname` "ignore"
- For every organizations that will be imported, we are cleaning the list of users and admins. Only the admins will be added as members of the wokspace.
- Only the Sterblue missions of the admins are to be imported as datasets in the workspace. There is a necessary deduplication step as admins often have access to the same sterblue missions

Some organizations were split so that not every members land in the same workspace with every datasets. Enedis for instance. So there should not be any permission problem.

**Comment by etiennedupont** [comment]
I don't get the deduplication part, are you talking about clones of missions?

**Comment by nicOwlas** [comment]
If user A has access to missions 1 and 2 and user B has access to missions 2 and 3, mission 2 should not be imported twice to the labelflow workspace.

**Comment by etiennedupont** [comment]
I was thinking about severity as well. Either we discard levels and anyway, they'll be available in the [backup](https://github.com/labelflow/labelflow/issues/492), or we can add them as separate label classes inside the name

**Comment by nicOwlas** [comment]
I'm for creating extra label classes. Some datasets would otherwise be useless. Corrosion on the australian transmission towers for instance.

For the public datasets we want to create there is a difficulty:  we are going to merge in the same dataset different missions having different anomalyLevels. We could use the universal referential for the class names and same for the levels? There was one universal anomalyLevelMapping per structure type?

**Comment by etiennedupont** [comment]
Let's maybe talk about that during the DSM, it's not very clear to me as of now

**Comment by nicOwlas** [comment]
@nacho-carnicero I've added to the description the query fields to put in the metadata field.

**Comment by etiennedupont** [comment]
@nicOwlas I guess we can close that one? 🎉
